### PR TITLE
Including tab for listing room uploaded files

### LINKED ISSUE #528: Files Uploaded to Channel on Right Panel
Having the ability to quickly see what files have been uploaded to the channel you're currently in on the right panel would be really good.

### PR DESCRIPTION
**Pull request by gmsecrieru**
This PR closes #528 and we can improve UI/UX should the need arise.

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
How hard would it be to have images pop up in the lightbox?

Right now you click an image it immediately downloads which is another deal all together.  :)

But looks very nice!

**Comment by gmsecrieru** [comment]
@geekgonecrazy Great catch! I will have a look at this.

File: server/publications/roomFiles.coffee [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  # list of channel messages which were created after uploading a file
+  msgQuery =
+    rid: rid
+    'file._id': { $exists: true }
+  msgOptions =
+    fields:
+      _id: 1
+      'file._id': 1
+    limit: 50
+  cursorFileMessages = RocketChat.models.Messages.find(msgQuery, msgOptions);
```
Comment (rodrigok) [inline server/publications/roomFiles.coffee]: Can you change this to use a named query, like `RocketChat.models.Messages.findMessagesWithFilesByRoomId`?

File: server/publications/roomFiles.coffee [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  uploadedFilesMessages = cursorFileMessages.fetch()
+  unless uploadedFilesMessages
```
Comment (rodrigok) [inline server/publications/roomFiles.coffee]: As far as i know the `fetch` always returns an array, so probably this `unless` will not work, right?

File: server/publications/roomFiles.coffee [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    return this.ready()
+
+  uploadIdList = _.map(uploadedFilesMessages, (doc) -> return doc?.file?._id)
+  unless uploadIdList
```
Comment (rodrigok) [inline server/publications/roomFiles.coffee]: Same problem here, _.map always returns an array

**Comment by gmsecrieru** [comment]
@rodrigok Thanks for your review! I will have a look at those as well.

**Comment by gmsecrieru** [comment]
@rodrigok suggested two great improvements:

* Add `rid` to the newly-created document during upload.
* Create a migration to handle current uploaded files.

I will work a bit more on this, thanks folks for the inputs!

**Comment by gmsecrieru** [comment]
I think that's it for now!

upload-swipebox

**Comment by geekgonecrazy** [comment]
This looks great! :+1:

**Comment by geekgonecrazy** [comment]
@gmsecrieru just pulled down on my production server.  Looks like images still download.  They show up in the preview but also still download.

**Comment by gmsecrieru** [comment]
I think I know what may have caused this. Will try something!

**Comment by geekgonecrazy** [comment]
@gmsecrieru I was trying it in the electron app.  Looks like it works fine in chrome.  I may need to tweak things on the desktop app

**Pull request merged**